main.go

```diff
@@ -1176,17 +1183,32 @@ func (tf *transformer) transformLinkname(localName, newName string) (string, str
 
 	var newForeignName string
 	if receiver, name, ok := strings.Cut(foreignName, "."); ok {
+
 		if lpkg.ImportPath == "reflect" && (receiver == "(*rtype)" || receiver == "Value") {
 			// These receivers are not obfuscated.
 			// See the TODO below.
 		} else if strings.HasPrefix(receiver, "(*") {
+
 			// pkg/path.(*Receiver).method
 			receiver = strings.TrimPrefix(receiver, "(*")
 			receiver = strings.TrimSuffix(receiver, ")")
-			receiver = "(*" + hashWithPackage(lpkg, receiver) + ")"
+
+			_, ok := tf.curPkgCache.ReflectObjects[pkgPath+"."+receiver]
+
+			if !ok {
+				receiver = hashWithPackage(lpkg, receiver)
+			}
+
+			receiver = "(*" + receiver + ")"
+
 		} else {
 			// pkg/path.Receiver.method
-			receiver = hashWithPackage(lpkg, receiver)
+
+			_, ok := tf.curPkgCache.ReflectObjects[pkgPath+"."+receiver]
+
+			if !ok {
+				receiver = hashWithPackage(lpkg, receiver)
+			}
```

Comment:
Move this logic into `hashWithPackage`, by also passing `tf` to it.

Use `lpkg.ImportPath` instead of `pkgPath`.

Reply:
I initially tried to use ImportPath, but that points to my package `handlers` rather than the golang ssh library path.

Reply:
Done as you said and it worked fine, so no idea why it didnt work for me initially.